src/ripple/app/misc/Validations.cpp

```diff
@@ -456,22 +465,27 @@ class ValidationsImp : public Validations
         if (mWriting)
             return;
 
-        mWriting = true;
-        app_.getJobQueue ().addJob (
-            jtWRITE, "Validations::doWrite",
-            [this] (Job&) { doWrite(); });
+        // addCountedJob() may return false (Job not added) at shutdown.
+        mWriting = app_.getJobQueue ().addCountedJob (
```

Comment:
Assigning `mWriting` the return value of addJob may be inaccurate. We may prefer to have `doWrite` be the only place that assigns `mWriting`. True upon entry and false on exit. That simplifies the code a bit and is precise on when we are writing.